devtools/datasette/publish.py

```diff
@@ -35,18 +35,27 @@
 
 DOCKERFILE_TEMPLATE = """
 FROM python:3.11.0-slim-bullseye
-COPY . /app
-WORKDIR /app
-
-RUN apt-get update
-RUN apt-get install -y zstd
 
-ENV DATASETTE_SECRET '{datasette_secret}'
 ENV DATABASES '{databases}'
+ENV DATASETTE_PORT 8081
+ENV NGINX_PORT 8080
+
+RUN apt-get update
+RUN apt-get install -y zstd nginx
 RUN pip install -U datasette datasette-cluster-map datasette-vega datasette-block-robots
-ENV PORT 8080
-EXPOSE 8080
 
+COPY nginx.conf /usr/share/nginx/nginx.conf
+COPY 50-mod-http-realip.conf /etc/nginx/modules-enabled/
+
+RUN mkdir /data \
+    && ln -sf /dev/stdout /var/log/nginx/access.log \
+    && ln -sf /dev/stderr /var/log/nginx/error.log
+
+COPY . /app
```

Comment:
Moved this down to the bottom since all the other setup changes much less frequently than the contents of the Docker context - this helps with build caching.